### ISSUE TITLE
Multiple entries in aircraft_modes

### ISSUE THREAD

**Issue by me2tkt** [Ysurac/FlightAirMap#261]
![aircraft_modes](https://cloud.githubusercontent.com/assets/23280102/21956764/18ed37f4-da88-11e6-9224-4f1758df8e78.png)

A sreenshot of my aircraft_modes db.
Why are there multiple entries for the same registration ?

**Comment by me2tkt** [comment]
I have been looking at these yesterday.
Some ModeS are for the previous aircraft with this registration.

For some gliders the last letter or number went missing eg HB-159 instead of HB-1593

There are a lot of entries for ModeS starting with DD or DDD, as far as I know these are not assigned to any country

So I am not sure if these errors are from the original data or the data import.
What is your source for the Basestation update ? 
I get mine from Planebase.

**Comment by Ysurac** [comment]
Planebase doesn't give a basestation anymore. Now I use multiple source like ACARS, and some national registry database.And I have a lot more data than latest available planebase basestation.
If there is an error, give me the **Source** and the error.
**DD** entry are used by gliders with Flarm.
**HB-159** was in flarm data file, I removed this DB in latest commit.
You can remove all data from flarm file like this:

    DELETE FROM aircraft_modes WHERE Source LIKE '%data.fln';

**Comment by me2tkt** [comment]
I am one of the early users of PlaneBase, so I still get the updates.
VirtualRadarServer and RadarCape seem to be able to access PlaneBase. But I dont know their arrangement.

I suspected that the DD where coming from Flarm as they were all gliders.

Should I post errors here ?
I try to cross check with airframes.org and other sites.

I am also slowly working my way through the airport table.
Adding city names, correcting the inconsistant spelling - but only for large and medium airports ;-)
e.g.  Johannesburg O.R. Tambo is now FAOR - not Orange River airport
however some airlines still use the old ICAO of FAJS (Jan Smuts)
I have added both to my database.
Particularly the writing for non English names

**Comment by Ysurac** [comment]
VirtualRadarServer ask planebase server for each **modeS**. They have a "special" access. Can't use that.
You can post error here, if they have **website_fam** for source and you are really sure it's an error.

For airports table I use wikipedia as source, so it's better to fix wikipedia (or tell me if wikipedia is right and not me, my script can have errors).

**Comment by me2tkt** [comment]
Are you using the Wikipedia "List of Airports by IATA and ICAO code"  or is there another table ?
I will put some examples here when I start again.

Radarcape downloads a new Basestation copy once a month but of course that is a commercial operation.

**Comment by Ysurac** [comment]
I only using wikipedia for coordinates and images... I need to fix that... I'm using wikipedia as source for airlines, but not yet for airports in fact. I will look at that asap.

I think the sources I use are quite correct and I will add more data soon.

**Comment by me2tkt** [comment]
I know it is difficult to get reliable data, routes are particularly bad.
Sometimes te aircraft is flying in a totally different direction.
But can also be due to wrong information sent from transponder due to crew error or laziness.

**Comment by Ysurac** [comment]
True. I don't have a good updated source for the info. Airlines alliance also don't give routes info anymore...

**Comment by me2tkt** [comment]
I prefer the flightroute.sqb files distributed via the PP-logs-and-routes Yahoo group by David J Taylor.
He provides an update about every 7 days
It seems to be more up to date than the route data provided by Andrew with his VirtualRadarServer.

You can find more details in this Yahoo group :
https://groups.yahoo.com/neo/groups/PP-logs-and-routes/info

However you would have to contact him as he has the following disclaimer
"This data may not be copied, published, or incorporated into other databases without the explicit permission"

It was originally designed for the Planeplotter programme but as it is in the sqlite format it can be accesed by any programme that can read sqlite files
